target/linux/realtek/files-6.12/drivers/net/phy/rtl83xx-phy.c

```diff
@@ -632,20 +637,41 @@ static int rtl8390_configure_generic(struct phy_device *phydev)
 	return 0;
 }
 
+static int rtl821x_prepare_patch(struct phy_device *phydev)
+{
+	int maxport = get_phys_in_package(phydev);
+	struct phy_device *patchphy;
+	int i, port, val;
+
+	for (port = 0; port < maxport; port++) {
+		patchphy = get_package_phy(phydev, port);
+		phy_write_paged(patchphy, RTL821X_PAGE_PATCH, 0x10, 0x10);
+	}
+	msleep(500);
+	for (port = 0; port < maxport; port++) {
+		patchphy = get_package_phy(phydev, port);
+		for (i = 0; i < 100 ; i++) {
+			val = phy_read_paged(patchphy, RTL821X_PAGE_STATE, 0x10);
+			if (val & 0x40)
+				break;
+		}
+		if (i >= 100) {
+			phydev_err(patchphy, "not ready for patch.\n");
+			return -1;
+		}
+	}
```

Comment:
This looks like you are polling for some value.

Do you rally have to sleep for 500ms or can you also start the polling immediately and wait for 10ms after each unsuccessful check? The you only wait as long as t is needed and still would also catch situations where it needs more than 500ms.